cg/cli/workflow/microsalt/base.py

```diff
@@ -10,149 +10,117 @@
-    elif sample_id and (order_id is None):
-        sample_obj = context.obj["db"].microbial_sample(sample_id)
-        if not sample_obj:
-            LOG.error("Sample %s not found", sample_id)
-            context.abort()
-        sample_objs = [context.obj["db"].microbial_sample(sample_id)]
-    elif sample_id and order_id:
-        microbial_order_obj = context.obj["db"].microbial_order(order_id)
-        if not microbial_order_obj:
-            LOG.error("Samples %s not found in %s ", sample_id, order_id)
-            context.abort()
-        sample_objs = [
-            sample_obj
-            for sample_obj in microbial_order_obj.microbial_samples
-            if sample_obj.internal_id == sample_id
-        ]
-    else:
-        LOG.error("provide order and/or sample")
+
+    if not ticket and not sample_id:
+        LOG.error("Provide ticket and/or sample")
+        context.abort()
+
+    sample_objs = context.obj["api"].get_samples(ticket, sample_id)
+
+    if not sample_objs:
+        LOG.error("No sample found for that ticket/sample_id")
         context.abort()
 
     parameters = [
         context.obj["lims_microsalt_api"].get_parameters(sample_obj) for sample_obj in sample_objs
     ]
 
-    filename = order_id if order_id else sample_id
-    outfilename = Path(context.obj["usalt"]["queries_path"]) / filename
-    outfilename = outfilename.with_suffix(".json")
-    if dry:
+    filename = str(ticket) if ticket else sample_id
+    outfilename = (Path(context.obj["usalt"]["queries_path"]) / filename).with_suffix(".json")
+    if dry_run:
         print(json.dumps(parameters, indent=4, sort_keys=True))
     else:
         with open(outfilename, "w") as outfile:
             json.dump(parameters, outfile, indent=4, sort_keys=True)
 
 
 @microsalt.command()
-@click.option("-d", "--dry", is_flag=True, help="print command to console")
-@click.option("-c", "--config-case", required=False, help="optionally change the config-case")
-@click.argument("order_id")
+@click.option("-d", "--dry-run", "dry_run", is_flag=True, help="print command to console")
+@click.option(
+    "-c",
+    "--config-case",
+    "config_case_path",
+    required=False,
+    help="optionally change the config-case",
+)
+@click.argument("ticket")
 @click.pass_context
-def run(context, dry, config_case, order_id):
+def run(context, dry_run, config_case_path, ticket):
     """ Start microSALT with an order_id """
+
     microsalt_command = context.obj["usalt"]["binary_path"]
     command = [microsalt_command]
 
-    config_case_path = config_case
-    if not config_case:
+    if not config_case_path:
         queries_path = Path(context.obj["usalt"]["queries_path"])
-        config_case_path = queries_path / order_id
-        config_case_path = config_case_path.with_suffix(".json")
+        config_case_path = (queries_path / ticket).with_suffix(".json")
 
     command.extend(["--parameters", str(config_case_path)])
-    if dry:
+    if dry_run:
```

Comment:
Flatten and use click.echo instead of print

Reply:
flattened